### PR TITLE
[ONNX] Add type conversion for Pow op inputs

### PR DESCRIPTION
**Pull request by mateusztabaka**
According to the OpenVino spec, both of the [Power ](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/arithmetic/Power_1.md) op inputs should have same type, but from [ONNX Pow](https://github.com/onnx/onnx/blob/master/docs/Operators.md#Pow) we can expect different type for data input and different for exponent input.
The goal is to insert Convert operation (if needed) to unify inputs type.

### CODE REVIEW

**Review by mitruska — CHANGES_REQUESTED** [review]
In general I have few suggestions:

- [ ] add `if` that check the element types are equal, if so no Convert insertion needed

- [ ] add Convert only for second input (power of the exponent)


- [ ] don't hardcode the `f32` type, use first input (data) `element_type` for the second input type conversion


- [ ] create `pow.cpp` and move the implementation there

File: ngraph/frontend/onnx_import/include/onnx_import/op/pow.hpp [hunk 1]
```diff
@@ -32,8 +32,12 @@ namespace ngraph
...
                 inline OutputVector pow(const Node& node)
                 {
-                    return {std::make_shared<default_opset::Power>(node.get_ng_inputs().at(0),
-                                                                   node.get_ng_inputs().at(1))};
+                    auto inputs = node.get_ng_inputs();
+                    return {std::make_shared<default_opset::Convert>(
+                        std::make_shared<default_opset::Power>(
+                            std::make_shared<default_opset::Convert>(inputs[0], element::f32),
+                            std::make_shared<default_opset::Convert>(inputs[1], element::f32)),
+                        inputs[0].get_element_type())};
```
Comment (mitruska) [inline ngraph/frontend/onnx_import/include/onnx_import/op/pow.hpp:40]: As in the general comment.

**Review by mitruska — CHANGES_REQUESTED** [review]
In general, I have few suggestions:

- [x] add `if` that check the element types are equal, if so no Convert insertion needed

- [ ] add Convert only for second input (power of the exponent)


- [ ] don't hardcode the `f32` type, use first input (data) `element_type` for the second input type conversion


- [ ] create `pow.cpp` and move the implementation there

File: ngraph/frontend/onnx_import/src/op/pow.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                    if (inputs[1].get_element_type() != base_type)
+                    {
+                        exponent = std::make_shared<default_opset::Convert>(inputs[1], base_type);
+                    }
+                    else
+                    {
+                        exponent = inputs[1].get_node_shared_ptr();
+                    }
+                    return {std::make_shared<default_opset::Power>(inputs[0], exponent)};
+                }
```
Comment (mitruska) [inline ngraph/frontend/onnx_import/src/op/pow.cpp:44]: I'm not sure this `else` with `exponent = inputs[1].get_node_shared_ptr();`, what do you think about small refactor:

```suggestion
OutputVector pow(const Node& node)
{
    auto inputs = node.get_ng_inputs();
    auto base_type = inputs[0].get_element_type();
    if (inputs[1].get_element_type() != base_type)
    {
        auto exponent = std::make_shared<default_opset::Convert>(inputs[1], base_type);
        return {std::make_shared<default_opset::Power>(inputs[0], exponent)};
    }
    return {std::make_shared<default_opset::Power>(inputs[0], inputs[1])};
}
```

**Review by postrational — APPROVED** [review]

File: ngraph/frontend/onnx_import/src/op/pow.cpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                                 inputs.size());
+
+                    auto base = inputs[0];
+                    auto exponent = inputs[1];
+                    auto base_type = inputs[0].get_element_type();
+                    auto exponent_type = inputs[1].get_element_type();
+                    if (exponent_type != base_type)
+                    {
+                        if (exponent_type.is_integral() ||
+                            (base_type.is_real() && base_type > exponent_type))
```
Comment (postrational) [inline ngraph/frontend/onnx_import/src/op/pow.cpp:45]: I'm a little concerned about this comparison.

Comment (mitruska) [reply ngraph/frontend/onnx_import/src/op/pow.cpp:45]: To not rely on the implementation of overloaded operators,
we can also use bitwidth so: `base_type.bitwidth() >= exponent_type.bitwidth()` 
What do you think?

**Review by lazarevevgeny — APPROVED** [review]

**Review by tomdol — APPROVED** [review]

**Review by mitruska — APPROVED** [review]

**Pull request merged**